documentation/private-link-service/private-link-redirect-map.md

```diff
@@ -0,0 +1,234 @@
+
+The entry is defined like below:
+
+```c
+typedef struct _sai_outbound_port_map_port_range_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Exact matched key outbound_port_map_id
+     *
+     * @objects SAI_OBJECT_TYPE_OUTBOUND_PORT_MAP
+     */
+    sai_object_id_t outbound_port_map_id;
+
+    /**
+     * @brief Range matched key dst_port_range
+     */
+    sai_u32_range_t dst_port_range;
+
+    /**
+     * @brief Rule priority in table
+     */
+    sai_uint32_t priority;
+
+} sai_outbound_port_map_port_range_entry_t;
+```
+
+The member `dst_port_range` designates the match field of port range, targeting packet tcp/udp destination port. The member `priority` is a selection hint if more than one entry is matched, and the higner number takes priority. The entry includes the following attributes:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_ACTION | sai_outbound_port_map_port_range_entry_action_t | Action `skip_mapping` or `map_to_private_link_service`. |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_IP | sai_ip_address_t | Back end IP, it overrides underlay destination IP |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_MATCH_PORT_BASE | sai_uint16_t | Match port base, aka begin port of port range |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_PORT_BASE | sai_uint16_t | Back end port base, aka begin port of translated port range |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_COUNTER_ID | sai_object_id_t | A counter that tracks the number of packets and bytes associated with the matching port range entry  |
+
+The entry attribute **Action** is defined as below:
+
+```c
+typedef enum _sai_outbound_port_map_port_range_entry_action_t
+{
+    SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ACTION_SKIP_MAPPING,
+    SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ACTION_MAP_TO_PRIVATE_LINK_SERVICE,
+} sai_outbound_port_map_port_range_entry_action_t;
+```
+
+The action `skip_mapping` means to skip the mapping operation. The action `map_to_private_link_service` means to do private link mapping operation as below:
+
+```c
+packet.underlay.destination_ip = entry.backend_ip
+packet.overlay.destination_port = entry.backend_port_base + (packet.overlay.destination_port - entry.match_port_base)
+```
+
+### 5.3. DASH CA-PA mapping attributes
+
+The following attributes will be added to CA-to-PA entry, for supporting service rewrites for PL/PL NSG redirect map:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_SVC_SRC_PREFIX | sai_ip_prefix_t | Service rewrite info - source ip prefix. |
+| SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_SVC_DST_PREFIX | sai_ip_prefix_t | Service rewrite info - destination ip prefix. |
+| SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_OUTBOUND_PORT_MAP_ID | sai_object_id_t | Outbound port map id. |
+
```

Comment:
We need 2 counters on ENI:
- Drop caused by port map missing
- Drop caused by port range entry missing